commit ac7c9291f72c6f54a3afb6268c312b106b07d525
Author: Dan Carpenter <dan.carpenter@oracle.com>
Date:   Tue Jul 10 20:32:51 2012 +0000

    smsc95xx: signedness bug in get_regs()
    
    "retval" has to be a signed integer for the error handling to work.
    
    Signed-off-by: Dan Carpenter <dan.carpenter@oracle.com>
    Signed-off-by: David S. Miller <davem@davemloft.net>

Index: linux-3.2.46/drivers/net/usb/smsc95xx.c
===================================================================
--- linux-3.2.46.orig/drivers/net/usb/smsc95xx.c	2013-07-26 19:37:23.000000000 +0000
+++ linux-3.2.46/drivers/net/usb/smsc95xx.c	2013-07-26 19:37:24.000000000 +0000
@@ -593,7 +593,8 @@
 			 void *buf)
 {
 	struct usbnet *dev = netdev_priv(netdev);
-	unsigned int i, j, retval;
+	unsigned int i, j;
+	int retval;
 	u32 *data = buf;
 
 	retval = smsc95xx_read_reg(dev, ID_REV, &regs->version);
